pkg/reader/ccip.go

```diff
@@ -628,6 +628,10 @@ func (r *ccipChainReader) discoverDestinationContracts(
 	}
 	for chain, cfg := range sourceConfigs {
 		resp = resp.Append(consts.ContractNameOnRamp, chain, cfg.OnRamp)
+		// The local router is located in each source chain config. Add it once.
+		if len(resp[consts.ContractNameOnRamp][r.destChain]) == 0 {
+			resp = resp.Append(consts.ContractNameRouter, r.destChain, cfg.Router)
+		}
```

Comment:
Previous versions of this PR had some complex logic dealing with the fact that there are multiple local router configurations. This has been simplified, we now take the first one and call it a day.